### PR TITLE
Add the primary detector info to the XRSTimeSeries if info exists in file

### PR DESCRIPTION
**Pull request by hayesla**
## PR Description

Fixes https://github.com/sunpy/sunpy/issues/6418

This PR adds a new column to the XRSTimeSeries when the xrsa_primary_chan and xrsb_primary_chan data exists within the file. These exist in the new GOES-R data (i.e. GOES16, 17 1s flux data), and tells you which primary detector is used that provides the flux values. This information is now needed for the temperature and emission measure calculations PR'd to sunkit-instruments - https://github.com/sunpy/sunkit-instruments/pull/81

### CODE REVIEW

**Comment by wtbarnes** [comment]
For our current XRS test data, how many of the possible code paths are we covering? i.e. do we have test data that covers all of the GOES satellites? And if not, can we simulate this in some way?

**Comment by hayesla** [comment]
yeah we definitely need to check that. I'll scope that out today and make sure our tests are covering all the files that XRSTimeSeries supports

**Comment by nabobalis** [comment]
I think this will need rebasing due to the new GOES test files?

File: sunpy/timeseries/sources/goes.py [hunk 1]
```diff
@@ -275,6 +280,14 @@ def _parse_netcdf(filepath):
                 ("xrsb_quality", u.dimensionless_unscaled),
             ]
         )
+        # add primary detector info if available
```
Comment (nabobalis) [inline sunpy/timeseries/sources/goes.py:283]:
```suggestion
        # Adds primary detector info
```

**Comment by nabobalis** [comment]
Needs a changelog

File: sunpy/timeseries/sources/goes.py [hunk 2]
```diff
@@ -245,7 +245,13 @@ def _parse_netcdf(filepath):
...
             start_time_str = start_time_str.lstrip("seconds since").rstrip("UTC").strip()
             times = Time(parse_time(start_time_str).unix + h5nc["time"], format="unix")
-
+            # check if the primary detector information exists
+            if "xrsa_primary_chan" in h5nc:
+                detector_info = True
+                xrsa_primary_chan = np.array(h5nc["xrsa_primary_chan"])
+                xrsb_primary_chan = np.array(h5nc["xrsb_primary_chan"])
+            else:
+                detector_info = False
```
[the remark this answers is not in the document]
Comment (hayesla) [reply sunpy/timeseries/sources/goes.py:254]: no they done luckily (we'll so far, we'll see about goes 18)

File: changelog/6454.feature.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Adds in abailabilyi to parse the primary detector information from the GOES-R XRS data if available to XRSTimeSeries.
```
Comment (nabobalis) [inline changelog/6454.feature.rst:1]:
```suggestion
`~.XRSTimeSeries` is now able to parse the primary detector information from the GOES-R XRS data if available.
```

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
Thanks for the changelog, now hpoefully someone else will review it.

**Review by dstansby — APPROVED** [review]

**Pull request merged**